internal/hcs/process.go

```diff
@@ -290,6 +290,12 @@ func (process *Process) StdioLegacy() (_ io.WriteCloser, _ io.ReadCloser, _ io.R
 		return nil, nil, nil, makeProcessError(process, operation, ErrAlreadyClosed, nil)
 	}
 
+	if process.stdin != nil && process.stdout != nil && process.stderr != nil {
+		in, out, err := process.stdin, process.stdout, process.stderr
+		process.stdin, process.stdout, process.stderr = nil, nil, nil
```

Comment:
Is there any kind of lock we should take here?

Reply:
I see `process.handleLock` is already used in most of the places where we use the stdio pipes, but it's probably clearer to leave that just for the process handle and add a new lock for the stdio pipes. the handle lock also currently only takes a reader lock in this case, which we'd need to upgrade to a writer lock if we wanted to reuse it.

Reply:
I've added locking around any usage of the stdio handles. I also set stdio handles to nil anywhere they are closed, to prevent double close or returning an invalid handle.